src/tests/ftest/util/ior_utils.py

```diff
@@ -1,186 +1,297 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+The Government's rights to use, modify, reproduce, release, perform, display,
+or disclose this software are subject to the terms of the Apache License as
+provided in Contract No. B609815.
+Any reproduction of computer software, computer software documentation, or
+portions thereof marked with this legend must also reproduce the markings.
+"""
 from __future__ import print_function
 
 import os
-import shutil
 import subprocess
 import json
-import sys
+
 
 class IorFailed(Exception):
-    """Raise if Ior failed"""
+    """Raise if Ior failed."""
 
-def get_ior_cmd(ior_flags, iteration, block_size, transfer_size, pool_uuid,
-                svc_list, record_size, stripe_size, stripe_count, async_io,
-                object_class, basepath, hostfile, proc_per_node=1, seg_count=1,
-                filename="`uuidgen`"):
-    """
-    Builds an IOR command given a set of input parameters, returns the command
-    as a string.  Some refactoring with run_ior will be done at a later date.
-    Consider this a WIP.
-    """
-    # some path wrangling
-    with open(os.path.join(basepath, ".build_vars.json")) as afile:
-        build_paths = json.load(afile)
-    orterun_bin = os.path.join(build_paths["OMPI_PREFIX"], "bin/orterun")
-    attach_info_path = basepath + "/install/tmp"
-    ior_bin = os.path.join(build_paths["OMPI_PREFIX"], "bin", "ior")
-    lib_path = os.path.join(build_paths["OMPI_PREFIX"], "lib")
-
-    # create the string containing the command
-    ior_cmd = (
-        orterun_bin + " --oversubscribe --mca mtl ^psm2,ofi "
-        "-x DAOS_SINGLETON_CLI -x OFI_INTERFACE -x LD_LIBRARY_PATH "
-        "-x CRT_ATTACH_INFO_PATH={1} -x CRT_PHY_ADDR_STR -x CRT_CTX_NUM "
-        "-x CRT_CTX_SHARE_ADDR {16} {2} -s {3} -i {4} -a DAOS -o {5} -b {6} "
-        "-t {7} --daos.pool={8} --daos.svcl={9} --daos.recordSize={10} "
-        "--daos.stripeSize={11} --daos.stripeCount={12} --daos.aios={13} "
-        "--daos.objectClass={14}".format(proc_per_node, attach_info_path,
-                                         ior_flags, seg_count, iteration,
-                                         filename, block_size, transfer_size,
-                                         pool_uuid, svc_list, record_size,
-                                         stripe_size, stripe_count, async_io,
-                                         object_class, hostfile, ior_bin,
-                                         lib_path))
-
-    return ior_cmd
+
+class IorParam(object):
```

Comment:
I think this could be abstracted to a file like: /daos/src/tests/ftest/util/cmd_utils.py   It would be beneficial since there are other cmdline tools (dmg cmdline, daos_server cmdline) that will end up using some form of what you did in the IorParam class. Just a thought since I'm using a lot of what you are doing to implement the ior cmdline to write tests for the dmg tool.

Reply:
After speaking with Maureen and Phil, I'll include a change for what I suggested on the patch that I will submit for the dmg tool test.